server/src/routes/api/v1/podcastVotesRouter.js

```diff
@@ -0,0 +1,105 @@
+import express from 'express'
+import objection from 'objection'
+const { ValidationError } = objection
+
+import { Vote } from '../../../models/index.js'
+
+const podcastVotesRouter = new express.Router({ mergeParams: true })
+
+podcastVotesRouter.get('/', async (req, res) => {
+  const { podcastId } = req.params
+  let userId
+  if (req.user) userId = req.user.id
+
+  try {
+    let votes = {}
+    votes.userVotes = await Vote.query()
+      .where('podcastId', podcastId)
+      .andWhere('userId', userId)
+      .first()
+    votes.totalVotes = await Vote.query()
+      .where('podcastId', podcastId)
+      .sum('value as value')
+      .first()
+    return res.status(201).json({ votes: votes })
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return res.status(422).json({ errors: error.data })
+    }
+    return res.status(500).json({ errors: error })
+  }
+})
+
+podcastVotesRouter.post('/', async (req, res) => {
+  const { value } = req.body
+  const { podcastId } = req.params
+  const userId = req.user.id
+
+  try {
+    let newVote = {}
+    newVote.userVotes = await Vote.query()
+      .insert({ podcastId, userId, value })
+      .first()
+      .returning('*')
+    newVote.totalVotes = await Vote.query()
+      .where('podcastId', podcastId)
+      .sum('value as value')
+      .first()
+    return res.status(201).json({ votes: newVote })
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return res.status(422).json({ errors: error.data })
+    }
+    return res.status(500).json({ errors: error })
+  }
+})
+
+podcastVotesRouter.patch('/', async (req, res) => {
+  const { value } = req.body
+  const { podcastId } = req.params
+  const userId = req.user.id
+
+  try {
+    let editVote = {}
+    editVote.userVotes = await Vote.query()
+      .update({ value: value })
+      .where('podcastId', podcastId)
+      .andWhere('userId', userId)
+      .first()
+      .returning('*')
+    editVote.totalVotes = await Vote.query()
+      .where('podcastId', podcastId)
+      .sum('value as value')
+      .first()
+    return res.status(201).json({ votes: editVote })
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return res.status(422).json({ errors: error.data })
+    }
+    return res.status(500).json({ errors: error })
+  }
+})
+
+podcastVotesRouter.delete('/', async (req, res) => {
+  const { podcastId } = req.params
+  const userId = req.user.id
+
+  try {
+    let deletedVote = {}
+    deletedVote.userVotes = await Vote.query().delete()
+      .where('podcastId', podcastId)
+      .andWhere('userId', userId)
+    deletedVote.totalVotes = await Vote.query()
+      .where('podcastId', podcastId)
+      .sum('value as value')
+      .first()
```

Comment:
same as previous routes
- no need for deletedVote, userVotes and totalVotes can exist on their own 
- I really dont see the need for userVotes at all in fact, since the return of delete is the number of deleted rows
- total votes makes sense, but should not be a property of deletedVote

Reply:
i just checked, and I dont even think you use `userVotes` after your DELETE fetch request at all